python/whylogs/experimental/api/logger/__init__.py

```diff
@@ -135,6 +135,26 @@ def _all_strings(data: pd.Series) -> bool:
     return all([all([isinstance(y, str) for y in x]) for x in data])
 
 
+def _get_segment_columns(schema: DatasetSchema, data: pd.DataFrame) -> List[str]:
+    columns: Set[str] = set()
+    for partition_name, partition in schema.segments.items():
+        if partition.filter:
```

Comment:
It looks like filters can access any column. I'm unclear if they're actually supported, but I think this should be safe.

Reply:
I think we should deprecate filter/query expression based segmentation. For ranking metrics segmentation support, since segment support in ranking metrics is new it would be better to throw an exception with message that filter segmentation is unsupported with ranking metrics, thoughts?

Reply:
It's not hard to support them, but I'm happy to veto them if they're deprecated...